### ISSUE TITLE
"prefer-const" ignores empty `let` declaration

### ISSUE THREAD

**Issue by Skalman** [eslint/eslint#13552]
**Tell us about your environment**

* **ESLint Version:** 7.6.0
* **Node Version:** eslint.org demo; 12.10.0
* **npm Version:** eslint.org demo; 6.10.3

**What parser (default, Babel-ESLint, etc.) are you using?** default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
{
    "parserOptions": {
        "ecmaVersion": 6,
        "sourceType": "script"
    },
    "rules": {
        "prefer-const": 2
    },
    "env": {
        "browser": true
    }
}
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
let a;
setTimeout(() => alert(a));
a = "abc";
```

```bash
npx eslint src/
```

**What did you expect to happen?**

No warning. `let a;` is just a shorter version of `let a = undefined;` which doesn't warn.

**What actually happened? Please include the actual, raw output from ESLint.**

```
/path/to/file.js
  1:5  error  'a' is never reassigned. Use 'const' instead  prefer-const

✖ 1 problem (1 error, 0 warnings)
```

**Are you willing to submit a pull request to fix this bug?**
Maybe

**Comment by mdjermanovic** [comment]
Hi @Skalman, thanks for the issue!

I think this is the intended default behavior of this rule.

You can set the option [ignoreReadBeforeAssign](https://eslint.org/docs/rules/prefer-const#ignorereadbeforeassign) to `true`:

```js
/* eslint prefer-const: ["error", { "ignoreReadBeforeAssign": true }] */ 

let a; // no errors
setTimeout(() => alert(a));
a = "abc";
```

[Online Demo](https://eslint.org/demo#eyJ0ZXh0IjoiLyogZXNsaW50IHByZWZlci1jb25zdDogW1wiZXJyb3JcIiwgeyBcImlnbm9yZVJlYWRCZWZvcmVBc3NpZ25cIjogdHJ1ZSB9XSAqLyBcblxubGV0IGE7IC8vIG5vIGVycm9yc1xuc2V0VGltZW91dCgoKSA9PiBhbGVydChhKSk7XG5hID0gXCJhYmNcIjsiLCJvcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYVZlcnNpb24iOjYsInNvdXJjZVR5cGUiOiJzY3JpcHQiLCJlY21hRmVhdHVyZXMiOnt9fSwicnVsZXMiOnt9LCJlbnYiOnt9fX0=)

**Comment by Skalman** [comment]
@mdjermanovic, you're correct. I should've read the docs more carefully - surprised that this isn't the default, but clearly others have different preferences. I'll fix my config :smile:
